src/evm-block-extractor/src/database/mod.rs

```diff
@@ -92,4 +94,41 @@ pub trait DatabaseClient: Send + Sync {
 
     /// Get earliest block number
     async fn get_earliest_block_number(&self) -> anyhow::Result<u64>;
+
+    /// Delete latest blocks starting with `start_from`, and related transactions.
+    /// Deleted blocks and transactions will be preserved in 'discarded' table with
+    /// the given 'reason' and timestamp.
+    async fn discard_blocks_starting_with(
+        &self,
+        start_from: u64,
+        reason: &str,
+    ) -> anyhow::Result<()>;
+
+    /// Returns a discarded block by its number.
+    async fn get_discarded_block_by_number(&self, number: u64) -> anyhow::Result<DiscardedBlock>;
+
+    /// Returns a discarded transaction from the database.
+    async fn get_discarded_transaction(&self, tx_hash: H256) -> anyhow::Result<Transaction>;
+
+    /// Returns block info from storage.
+    ///
+    /// # Warning
+    /// Do not use this info fields as indexes for blocks in storage.
+    /// The following numbers are about block numbers in the source blockchain
+    /// and can exceed the latest block number in the database.
+    /// - latest_block_number: u64,
+    /// - safe_block_number: u64,
+    /// - finalized_block_number: u64,
+    /// - pending_block_number: u64,
+    async fn get_block_info(&self) -> anyhow::Result<Option<BlockchainBlockInfo>>;
+
+    /// Stores blockchain block info.
+    async fn set_block_info(&self, info: BlockchainBlockInfo) -> anyhow::Result<()>;
+}
+
+#[derive(Debug)]
+pub struct DiscardedBlock {
+    pub block: Block<H256>,
+    pub reason: String,
+    pub timestamp: chrono::NaiveDateTime,
```

Comment:
why `NaiveDateTime`? It should be `DateTime` instead

Reply:
I think chrono::DateTime<Utc> is probably the best option

Reply:
Fixed